CRM/Contribute/Form/ContributionRecur.php

```diff
@@ -0,0 +1,537 @@
+      CRM_Custom_Form_CustomData::buildQuickForm($this);
+      CRM_Custom_Form_CustomData::setDefaultValues($this);
+    }
+
+    parent::preProcess();
+  }
+
+  function setDefaultValues() {
+    if ($this->_cdType) {
+      return CRM_Custom_Form_CustomData::setDefaultValues($this);
+    }
+
+    $defaults = $this->_values;
+
+    if ($this->_id) {
+      $this->_contactID = $defaults['contact_id'];
+    }
+
+    if (isset($defaults['total_amount'])) {
+      $defaults['amount'] = CRM_Utils_Money::format($defaults['amount'], NULL, '%a');
+    }
+
+    $dates = array('start_date', 'end_date', 'cancel_date', 'next_sched_contribution_date');
+    foreach ($dates as $key) {
+      if (!empty($defaults[$key])) {
+        list($defaults[$key],
+          $defaults[$key . '_time']
+          ) = CRM_Utils_Date::setDateDefaults(CRM_Utils_Array::value($key, $defaults),
+          'activityDateTime'
+        );
+      }
+    }
+    
+    if ($this->_action & CRM_Core_Action::DELETE) {
+      return $defaults;
+    }
+    
+    // Set move existing contributions to TRUE as default
+    $defaults['move_existing_contributions'] = 1;
+    $defaults['contact_id'] = $this->_contactID;
+    $defaults['selected_cid'] = $this->_contactID;
+
+    $this->_defaults = $defaults;
+    return $defaults;
+  }  
+    
+  /**
+   * Build the form
+   *
+   * @access public
+   * @return void
+   */
+  function buildQuickForm( ) {
+
+    if ($this->_action & CRM_Core_Action::DELETE) {
+      // Check if any contributions created for the recurring record
+      $recurringIds = array($this->_id);
+      $contributionCount = CRM_Contribute_BAO_ContributionRecur::getCount($recurringIds);
+      if (isset($contributionCount[$this->_id]) && $contributionCount[$this->_id] > 0) {
+        $this->assign('contributionCount' , $contributionCount[$this->_id]);
+      }
+      $this->addButtons(array(
+          array(
+            'type' => 'next',
+            'name' => ts('Delete'),
+            'spacing' => '&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;',
+            'isDefault' => TRUE,
+          ),
+          array(
+            'type' => 'cancel',
+            'name' => ts('Cancel')
+          )
+        )
+      );
+      return;
+    }
+    
+    if ($this->_cdType) {
+      CRM_Custom_Form_CustomData::buildQuickForm($this);
+      return;
+    }
+
+    //need to assign custom data type to the template
+    $this->assign('customDataType', 'ContributionRecur');
+    $this->assign('entityID', $this->_id);
+    
+    $attributes = CRM_Core_DAO::getAttribute('CRM_Contribute_DAO_ContributionRecur');
+
+    $cid        = CRM_Utils_Request::retrieve('cid',  'Integer', $this);
+    $id         = CRM_Utils_Request::retrieve('crid', 'Integer', $this);
+
+    $paymentProcessor = $this->add('select', 'payment_processor_id',
+      ts('Payment Processor'),
+      array('' => ts('- select -')) + CRM_Contribute_BAO_ContributionRecur::getBackOfficePaymentProcessors(),
```

Comment:
per previous comments - try to use the methods the other forms are now consolidated as using